modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/ParameterTypeModelValidator.java

```diff
@@ -39,12 +43,35 @@ public void validate(ExtensionModel extensionModel, ProblemsReporter problemsRep
 
       @Override
       protected void onParameter(ParameterizedModel owner, ParameterGroupModel groupModel, ParameterModel model) {
-        validateParameter(model, problemsReporter);
+        validateParameterType(model, problemsReporter);
+        validateParameterWithReferencesType(model, problemsReporter);
       }
     }.walk(extensionModel);
   }
 
-  private void validateParameter(ParameterModel parameter, ProblemsReporter problemsReporter) {
+  private void validateParameterWithReferencesType(ParameterModel parameter, ProblemsReporter problemsReporter) {
+    if (!parameter.getElementReferences().isEmpty()) {
+      parameter.getType().accept(new MetadataTypeVisitor() {
```

Comment:
This can be simply done with `if(!(parameter.getType() instanceOf StringType)) addError`